### PR TITLE
feat(web): key-event rule processing unit tests for mnemonic, KMW 1.0 keyboards 🧓

### PR DESCRIPTION
**Pull request by jahorton**
This PR adds unit tests to ensure that rule processing for mnemonic keyboards and positional keyboards operates correctly given their respective key-event format specifications as tested by unit tests introduced in #10464. 

This does _not_ test that the key-event objects input to the rules are properly constructed from hardware keystrokes to match the active keyboard.  That's the responsibility of #10464's unit tests.  Here, assuming the input is supplied properly, we test that the rules are applied properly.

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/keyboard-processor/tests/node/non-positional-rules.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      assert.isOk(legacyResult);
+      assert.isFalse(legacyResult.triggerKeyDefault);
+      assert.equal(mockLegacy.getText(), 'ա');
+    });
+
+    it('does not match rules with mnemonic-specced KeyEvents', () => {
+      // Note:  plain 'n' is produced from default key outputs for sil_ipa, not a keyboard rule.
+      let mockMnemonic = new Mock('');
+      let mnemonicEvent = new KeyEvent({
+        // sil_ipa is a mnenomic keyboard:  it expects codes based on the key's standard character output.
```
Comment (ermshiperete) [inline common/web/keyboard-processor/tests/node/non-positional-rules.js:166]:
```suggestion
        // s armenian is a KMW 1.0 keyboard:  it expects codes based on the key's standard character output.
```

**Review by ermshiperete — APPROVED** [review]

File: common/web/keyboard-processor/tests/node/non-positional-rules.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+const require = createRequire(import.meta.url);
+
+import { Codes, KeyboardInterface, KeyEvent, MinimalKeymanGlobal, Mock } from '@keymanapp/keyboard-processor';
+import { NodeKeyboardLoader } from '@keymanapp/keyboard-processor/node-keyboard-loader';
+
+// Compare and contrast the unit tests here with those for app/browser key-event unit testing
+// in the hardware-event-processing set; the output objects there should have the same format
+// as the event object inputs used here.
+
+describe.only('Engine - rule processing', function() {
```
[the remark this answers is not in the document]
Comment (jahorton) [reply common/web/keyboard-processor/tests/node/non-positional-rules.js:14]:
```suggestion
describe('Engine - rule processing', function() {
```

**Comment by jahorton** [comment]
Huh, this one's new to me:

```
10:57:00 
  *** Building scheme "XCGLogger (iOS)" in XCGLogger.xcodeproj

10:57:18 
  *** Building scheme "DeviceKit" in DeviceKit.xcworkspace

10:57:34 
  *** Building scheme "Reachability" in Reachability.xcodeproj

11:00:58 
  xcodebuild timed out while trying to read iOS-ObjectiveC.xcodeproj

11:00:58 
  

11:00:58 
  [ios/engine] Carthage dependency loading failed

11:00:58 
  

11:00:58 
  [ios/engine] ## configure failed
```

**Comment by mcdurdin** [comment]
> Huh, this one's new to me:

This is what we were trying to diagnose yesterday with the iOS failing builds. See #10467.

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.253-alpha](https://keyman.com/downloads/releases/alpha/17.0.253)

**Pull request merged**